### DISCUSSION TITLE
How does this compare to Missionary?

### DISCUSSION THREAD

**Discussion by kxygk** [lilactown/flex#3] [General]
This library seems quite minimal and nice. It also seems to be doing something quite similar to Missionary - and I don't feel I fully appreciate the difference (flex's interface looks very simple and limited in scope in comparison).  I was just wondering how you saw flex vis a vis Missionary and if you had any general thoughts

It's been a real struggle for me to get anything going with Missionary - there is definitely a learning curve and I don't fully grok how to work with it. For little programming projects where I'm not doing anything too crazy it seems to end up needing more machinery than something like `flex`. That said it conceptually looks very interesting and I wish I understood it better. They seem to be really solving some highly complex networking synchronization problems. But that's just my halfbaked high level impression

**Comment by lilactown** [comment]
I've had a similar experience with missionary. I've attempted to learn it a few times and each time bounced off the syntax and jargon it uses. I can't give a detailed comparison because of that. I can tell you that missionary seems very general and quite capable given what the author and others are building over at [electric](https://github.com/hyperfiddle/electric). Perhaps that is the best way to consume it.

For me, flex meets my personal aesthetic sensibilities and needs for building UIs. It draws from a rich conceptual and syntactic history in many other programming languages. It is meant to be a relatively focused, but powerful, library.

Maybe someone else who has experience in missionary can give flex a fair shake and do a deep comparison of the two.

**Reply by kxygk** [reply]
okay, thanks for the sanity check :))

I've been looking at theirs docs and exchanging some messages with the maintainer and was feeling a bit stupid for still not really getting it. Since the maintainers seem quite responsive, I might try to build a minimal flex example and then see if they can help convert it to missionary to see how it'd look
